History Get function added

iepathos opened the pull request:
Added a history.Get function to return x lines back in the history as a string slice.  I find this func useful for dealing with history in the command shell.  I was looping through older/newer before adding this and it simplified things a lot for me.  Added a test to make sure it's working as expected.

c-bata commented on history.go:
```diff
@@ -51,6 +52,14 @@ func (h *History) Newer(buf *Buffer) (new *Buffer, changed bool) {
 }
 
+// Get x lines back in the history as a string array
+func (h *History) Get(lines int) []string {
```
[imo] I have a little concerned about this method name/

* We don't imagine that this method returns the histories from the last? or from we selected?
* When we want to add another method to retrieve *one* history, this name will conflict.

c-bata commented:
Thanks for submitting PR.
I left some comments about your changes. Anyway, I'm sure this PR will be merged 👍

iepathos commented:
Hello, added a test with the panic case from above and fixed the error.  I also updated the function naming here a bit so it's less likely to hit a future conflict.  Changed function Get to GetLines, added a GetLine and GetLast to go with it and tests for them as well.

c-bata commented on history.go:
```diff
@@ -51,6 +51,33 @@ func (h *History) Newer(buf *Buffer) (new *Buffer, changed bool) {
+    if begin > h.selected - 1 {
+    	begin = 0
+    }
+	return h.histories[begin:h.selected]
```
It seems indents are not aligned.

c-bata commented on history.go:
```diff
@@ -51,6 +51,33 @@ func (h *History) Newer(buf *Buffer) (new *Buffer, changed bool) {
 	return new, true
 }
 
+// Get x lines back in the history as a string array
```
Godoc comments should start function name like `GetLines returns ....` .
https://blog.golang.org/godoc-documenting-go-code
